File: JavaScript/2dArrayQuestions/pascalsTriangle.js
function generatePascalsTriangle(numRows) {
    const triangle = [];
    
    for (let rowIndex = 0; rowIndex < numRows; rowIndex++) {
        const row = Array(rowIndex + 1);
        
        for (let j = 0; j <= rowIndex; j++) {
            if (j === 0 || j === rowIndex) {
                row[j] = 1;
            } else {
                row[j] = triangle[rowIndex - 1][j - 1] + triangle[rowIndex - 1][j];
            }
        }
        triangle.push(row);
    }
    
    return triangle;
}

console.log('Pascal\'s Triangle:', generatePascalsTriangle(5));

File: src/pascalsTriangle.java
public class pascalsTriangle {
    public static void main(String[] args) {
        int numRows = 5;

        int[][] triangle = generatePascalsTriangle(numRows);

        printTriangle(triangle);
    }

    public static int[][] generatePascalsTriangle(int numRows) {
        int[][] triangle = new int[numRows][];

        for (int i = 0; i < numRows; i++) {
            triangle[i] = new int[i + 1];
            triangle[i][0] = 1;
            triangle[i][i] = 1;

            for (int j = 1; j < i; j++) {
                triangle[i][j] = triangle[i - 1][j - 1] + triangle[i - 1][j];
            }
        }

        return triangle;
    }

    public static void printTriangle(int[][] triangle) {
        for (int[] row : triangle) {
            for (int num : row) {
                System.out.print(num + " ");
            }
            System.out.println();
        }
    }
}
